Makefile

```diff
@@ -1,11 +1,76 @@
 include .bingo/Variables.mk
 
-.DEFAULT_GOAL=textile
+.DEFAULT_GOAL=install
 
-GOVVV_FLAGS=$(shell $(GOVVV) -version git -flags -pkg $(shell go list ./buildinfo))
+TXTL_BUILD_FLAGS?=CGO_ENABLED=0
+TXTL_VERSION?="none"
+GOVVV_FLAGS=$(shell $(GOVVV) -flags -version $(TXTL_VERSION) -pkg $(shell go list ./buildinfo))
 
-textile: $(GOVVV)
-	go install -ldflags="${GOVVV_FLAGS}" ./...
+build: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go build -ldflags="${GOVVV_FLAGS}" ./...
+.PHONY: build
+
+build-hub: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go build -ldflags="${GOVVV_FLAGS}" ./cmd/hub
+.PHONY: build-hub
+
+build-hubd: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go build -ldflags="${GOVVV_FLAGS}" ./cmd/hubd
+.PHONY: build-hubd
+
+build-buck: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go build -ldflags="${GOVVV_FLAGS}" ./cmd/buck
+.PHONY: build-buck
+
+build-buckd: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go build -ldflags="${GOVVV_FLAGS}" ./cmd/buckd
+.PHONY: build-buckd
+
+install: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go install -ldflags="${GOVVV_FLAGS}" ./...
+.PHONY: install
+
+install-hub: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go install -ldflags="${GOVVV_FLAGS}" ./cmd/hub
+.PHONY: install-hub
+
+install-hubd: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go install -ldflags="${GOVVV_FLAGS}" ./cmd/hubd
+.PHONY: install-hubd
+
+install-buck: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go install -ldflags="${GOVVV_FLAGS}" ./cmd/buck
+.PHONY: install-buck
+
+install-buckd: $(GOVVV)
+	$(TXTL_BUILD_FLAGS) go install -ldflags="${GOVVV_FLAGS}" ./cmd/buckd
+.PHONY: install-buckd
+
+define gen_release_files
+	$(GOX) -osarch=$(3) -output="build/$(2)/$(2)_${TXTL_VERSION}_{{.OS}}-{{.Arch}}/$(2)" -ldflags="${GOVVV_FLAGS}" $(1)
+	mkdir -p build/dist; \
+	cd build/$(2); \
+	for release in *; do \
+		cp ../../LICENSE ../../README.md $${release}/; \
+		if [[ $${release} != *"windows"* ]]; then \
+  		TXTL_FILE=$(2) $(GOMPLATE) -f ../../dist/install.tmpl -o "$${release}/install"; \
+			tar -czvf ../dist/$${release}.tar.gz $${release}; \
+		else \
+			zip -r ../dist/$${release}.zip $${release}; \
+		fi; \
+	done
+endef
+
+build-hub-release: $(GOX) $(GOVVV) $(GOMPLATE)
+	$(call gen_release_files,./cmd/hub,hub,"linux/amd64 linux/386 linux/arm darwin/amd64 windows/amd64")
+.PHONY: build-hub-release
+
+build-buck-release: $(GOX) $(GOVVV) $(GOMPLATE)
+	$(call gen_release_files,./cmd/buck,buck,"linux/amd64 linux/386 linux/arm darwin/amd64 windows/amd64")
+.PHONY: build-buck-release
+
+build-releases: build-hub-release build-buck-release
+.PHONY: build-releases
```

Comment:
We can add releases for `hubd` and `buckd` as well if we want, it's now trivial. Didn't do that because I figured there is a reason we're not releasing them?

Reply:
Ah yeah, I think we decided to not promote the use of `hubd` but doing that by not releasing the artifacts doesn't sound right anymore. I vote to include them.

Reply:
Ok will do.